File: 1149_RGB거리/rgb.js
/*

2017 Dec 16, made by Moon

*/

function main(c){
	var val, d = [];
  var ix, ixLen;
  d[0] = 0;

  for(ix = 1, ixLen = c.length; ix < ixLen; ix++){
  	d[ix] = new Array();
    if(ix === 1){
    	d[ix][0] = c[ix][0];
      d[ix][1] = c[ix][1];
      d[ix][2] = c[ix][2];
    }else{
    	d[ix][0] = Math.min(d[ix-1][1], d[ix-1][2]) + c[ix][0];
      d[ix][1] = Math.min(d[ix-1][0], d[ix-1][2]) + c[ix][1];
      d[ix][2] = Math.min(d[ix-1][0], d[ix-1][1]) + c[ix][2];
    }
  }

  console.log(Math.max(d[ix-1][0], d[ix-1][1], d[ix-1][2]));
}

main([
	[26, 40, 83],
  [49, 60, 57],
  [13, 89, 99]
])

File: 1149_RGB거리/rgb.c
/*

2017 Dec 20, made by Jeon

*/

RGB거리 

#include <stdio.h>
#define N 1001
#define MIN(A,B) A>B?B:A

int main(void)
{
	int n, temp;
	int R[N] = { 0 };
	int G[N] = { 0 };
	int B[N] = { 0 };
	scanf("%d", &n);
	
	for (int i = 1; i <= n; i++)
	{
		int cr, cg, cb;
		scanf("%d %d %d", &cr, &cg, &cb);

		R[i] = MIN(G[i - 1] + cr, B[i - 1] + cr);
		G[i] = MIN(R[i - 1] + cg, B[i - 1] + cg);
		B[i] = MIN(R[i - 1] + cb, G[i - 1] + cb);
	}

	temp = MIN(R[n], G[n]);
	printf("%d\n", MIN(temp, B[n]));

	return 0;
}